### PR TITLE
Add Custom .NET CLI support to OmniSharp

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Add a configuration mechanism to allow users to point OmniSharp to a custom .NET CLI, with a fallback to the DOTNET_ROOT environment variable before falling back to the PATH.

I've updated the tests to use this functionality for locating the .NET CLI instead of using the old test-only code-path.

As this is my first contribution to Omnisharp, I'm not quite sure how this exposes the new setting through to omnisharp.json, so if anyone has any pointers on that, they would be well appreciated.

cc: @333fred

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
cc: @filipw can you take a look at this PR when you have a chance?

**Review by 333fred — APPROVED** [review]

**Review by filipw — APPROVED** [review]
LGTM - thanks!

**Comment by SamB** [comment]
FWIW, it kind of looks like this runs before the omnisharp.json files are loaded. ☹

File: src/OmniSharp.Host/Services/DotNetCliService.cs [hunk 1]
```diff
@@ -20,14 +25,45 @@ internal class DotNetCliService : IDotNetCliService
...
 
-            DotNetPath = dotnetPath ?? "dotnet";
+            // Check if any of the provided paths have a dotnet executable.
+            string executableExtension = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".exe" : string.Empty;
+            foreach (var path in dotNetCliOptions.Value.GetNormalizedLocationPaths(environment))
+            {
+                if (File.Exists(Path.Combine(path, $"dotnet{executableExtension}")))
+                {
+                    // We'll take the first path that has a dotnet executable.
+                    DotNetPath = Path.Combine(path, "dotnet");
```
[the remark this answers is not in the document]
Comment (jkoritzinsky) [reply src/OmniSharp.Host/Services/DotNetCliService.cs:42]: I think you're right, so a PR is probably worthwhile.

**Pull request merged**